### PR TITLE
Use Expo for handling built-in plural forms

### PR DESCRIPTION
**Pull request by whatyouhide**

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -49,7 +49,8 @@ defmodule Gettext.Mixfile do
 
   defp deps do
     [
-      {:expo, "~> 0.1.0 or ~> 0.2.0"},
+      # {:expo, "~> 0.1.0 or ~> 0.2.0"},
+      {:expo, github: "elixir-gettext/expo"},
```
Comment (whatyouhide) [inline mix.exs:53]: We'll update this once we release a new Expo version.

File: lib/gettext/plural.ex [hunk 2]
```diff
@@ -256,282 +256,25 @@ defmodule Gettext.Plural do
...
-    with "nplurals=" <> rest <- String.trim(plural_forms_header),
-         {plural_forms, _rest} <- Integer.parse(rest) do
-      {locale, plural_forms}
-    else
-      _other -> locale
+    case Expo.PluralForms.parse(plural_forms_header) do
+      {:ok, plural_forms} ->
+        {locale, plural_forms}
+
+      {:error, _reason} ->
```
Comment (maennchen) [inline lib/gettext/plural.ex:269]: @whatyouhide Is a plural forms header with only the nplurals part technically valid? If yes we should maybe parse it to a plural forms struct with a nil plural field.

**Review by maennchen — APPROVED** [review]

File: lib/gettext/plural.ex (hunk 2 above)
Comment (maennchen) [reply lib/gettext/plural.ex:269]: I'm trying to find an answer to this in the official gettext source.

File: lib/gettext/plural.ex [hunk 3]
```diff
@@ -256,282 +256,25 @@ defmodule Gettext.Plural do
...
+        # Fall back to parsing headers such as "nplurals=3", without the "plural=..." part.
+        # TODO: remove this at some point.
+        with "nplurals=" <> rest <- String.trim(plural_forms_header),
+             {plural_forms, _rest} <- Integer.parse(rest) do
+          {locale, plural_forms}
```
Comment (maennchen) [inline lib/gettext/plural.ex:274]: Would it make sense to put something along the lines of this (untested pseudo-code):

```suggestion
          case Expo.PluralForms.plural_form(locale) do
            {:ok, plural_form} -> IO.warn("""
            Deprecated `Plural-Forms` header found:
            Current: #{inspect(plural_forms_header)}
            Recommendation: #{inspect(Expo.PluralForms.to_string(plural_form))}
            """)
            :error -> :ok
          end
          {locale, plural_forms}
```

File: lib/gettext/plural.ex (hunk 2 above)
Comment (maennchen) [reply lib/gettext/plural.ex:269]: I can't find anything. Let's keep it as is.

Comment (whatyouhide) [reply lib/gettext/plural.ex:269]: @maennchen yeah I don't think it's valid and I'd rather err on the side of not being valid, so I hacked this together to support it only in Gettext for now (and not in Expo). We can remove this support later on 🙃

File: lib/gettext/plural.ex (hunk 3 above)
Comment (whatyouhide) [reply lib/gettext/plural.ex:274]: Yes! I'll do this in another PR, where we can deprecate `--plural-forms` as well and so on.

**Pull request merged**